wasmtime/_memory.py

```diff
@@ -62,6 +62,73 @@ def data_ptr(self, store: Storelike) -> "ctypes._Pointer[c_ubyte]":
         """
         return ffi.wasmtime_memory_data(store._context, byref(self._memory))
 
+    def set_store(self, store: Storelike):
+        """
+        must be called to set the store for highlevel access of memory
+        """
+        self._store=store
```

Comment:
I think it would be better to not have this form of storage and take the store as input to the methods below. Additionally could they perhaps be renamed to `read` and `write` to be similar to the Wasmtime embedding and convey that copies are being performed as opposed to acquiring views?